### ISSUE TITLE
Cannot supply `defaults` object to Model.findOrCreate

### ISSUE THREAD

**Issue by samcal** [RobinBuschmann/sequelize-typescript#94]
I'm not sure if this is a bug with `sequelize-typescript`, or with the Typescript definitions, but I get a compile-time error if I try to use the `defaults` option for `findOrCreate`:

```typescript
await WaitlistUser.findOrCreate<WaitlistUser>({
  where: {
    email
  },
  defaults: {
    enteredWaitlistAt: new Date(),
    status: 'new'
  }
})
```

But this doesn't typecheck, since the compiler is expecting `defaults` to be of type `WaitlistUser`, and the `defaults` object doesn't have an `id` key. My workaround for the moment is at add an `as any` annotation to the `defaults` object.

**Comment by RobinBuschmann** [comment]
Hey @samcal which sequelize-typescript version are you using?

**Comment by samcal** [comment]
I'm using version 0.3.5, I can update to 0.4.0 if you think that will help

**Comment by RobinBuschmann** [comment]
Yes, this will help. The fix is part of the `0.4.0` release (https://github.com/RobinBuschmann/sequelize-typescript/pull/64/files#diff-1c485f95a55d5ce0466f63a966ced297R330)

**Comment by samcal** [comment]
Perfect, thanks so much! I should have checked before opening the issue.

Thanks for the great work on this repo, makes my job much easier 😃
